src/Repository/CompanyRepository.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace App\Repository;
+
+use App\Entity\Company;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Persistence\ManagerRegistry;
+
+/**
+ * @extends ServiceEntityRepository<Company>
+ *
+ * @method Company|null find($id, $lockMode = null, $lockVersion = null)
+ * @method Company|null findOneBy(array $criteria, array $orderBy = null)
+ * @method Company[]    findAll()
+ * @method Company[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+ */
+class CompanyRepository extends ServiceEntityRepository
+{
+    /**
+     * @param ManagerRegistry $registry
+     */
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, Company::class);
+    }
+
+    /**
+     * @param Company $entity
+     * @return void
+     */
+    public function save(Company $entity): void
+    {
+        $this->getEntityManager()->persist($entity);
+        $this->getEntityManager()->flush();
+    }
+
+    /**
+     * @param Company $entity
+     * @return void
+     */
+    public function remove(Company $entity): void
+    {
+        $this->getEntityManager()->remove($entity);
+        $this->getEntityManager()->flush();
+    }
+
+    /**
+     * @param int $id
+     * @return Company|null
+     */
+    public function removeById(int $id): string
+    {
+        $listOfCompanies = $this->findAll();
```

Comment:
No meed to get all companies, you can replace with ->find($id) as the id is unique being a Primary Key we will do max 1 delete. Your approach is good when we need to delete all companies that have a specific name because we can have more companies with the same name. In that case we will do multiple deletes.